index.js

```diff
@@ -165,119 +166,100 @@ EventEmitter.prototype.emit = function emit(event, a1, a2, a3, a4, a5) {
 
 /**
  * Add a listener for a given event.
- *
+ * 
+ * @param {EventEmitter} `this`
  * @param {String|Symbol} event The event name.
  * @param {Function} fn The listener function.
  * @param {Mixed} [context=this] The context to invoke the listener with.
+ * @param {Boolean} [once] Whether a one-time listener
  * @returns {EventEmitter} `this`.
- * @api public
+ * @api private
  */
-EventEmitter.prototype.on = function on(event, fn, context) {
-  var listener = new EE(fn, context || this)
+function _addListener(emitter, event, fn, context, once) {
+  var listener = new EE(fn, context || emitter, once)
     , evt = prefix ? prefix + event : event;
 
-  if (!this._events[evt]) this._events[evt] = listener, this._eventsCount++;
-  else if (!this._events[evt].fn) this._events[evt].push(listener);
-  else this._events[evt] = [this._events[evt], listener];
+  if (!emitter._events[evt]) emitter._events[evt] = listener, emitter._eventsCount++;
+  else if (!emitter._events[evt].fn) emitter._events[evt].push(listener);
+  else emitter._events[evt] = [emitter._events[evt], listener];
 
-  return this;
+  return emitter;
+}
+
+EventEmitter.prototype.on = function on(event, fn, context) {
+  return _addListener(this, event, fn, context, false);
 };
 
-/**
- * Add a one-time listener for a given event.
- *
- * @param {String|Symbol} event The event name.
- * @param {Function} fn The listener function.
- * @param {Mixed} [context=this] The context to invoke the listener with.
- * @returns {EventEmitter} `this`.
- * @api public
- */
 EventEmitter.prototype.once = function once(event, fn, context) {
-  var listener = new EE(fn, context || this, true)
-    , evt = prefix ? prefix + event : event;
-
-  if (!this._events[evt]) this._events[evt] = listener, this._eventsCount++;
-  else if (!this._events[evt].fn) this._events[evt].push(listener);
-  else this._events[evt] = [this._events[evt], listener];
-
-  return this;
+  return _addListener(this, event, fn, context, true);
 };
 
 /**
  * Remove the listeners of a given event.
  *
- * @param {String|Symbol} event The event name.
+ * @param {String|Symbol} [event] The event name.
  * @param {Function} fn Only remove the listeners that match this function.
  * @param {Mixed} context Only remove the listeners that have this context.
  * @param {Boolean} once Only remove one-time listeners.
  * @returns {EventEmitter} `this`.
  * @api public
  */
 EventEmitter.prototype.removeListener = function removeListener(event, fn, context, once) {
-  var evt = prefix ? prefix + event : event;
+  var evt;
 
-  if (!this._events[evt]) return this;
-  if (!fn) {
+  function clearEvent() {
```

Comment:
Try to move this function outside the `removeListener` method, and call it without using `Function.prototype.call()` as you did above.